### PR TITLE
Update releasing instructions with latest infos

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -182,46 +182,33 @@ First:
...
-
-Then:
 
-The "Back to development" commit cannot be pushed to main because we don't allow pushes to main.
-
-You can just throw away the commit (``git reset --hard HEAD^``) but
-the next person to touch the changelog will have to introduce a new
-heading for the next version. Another option is to push the commit and
-have it be reviewed:
+- At this point the package is published on Pypi. Now prepare the next version and push the tag to the repo with:
```
Comment (grahamalama) [inline README.rst:193]: > At this point the package is published on Pypi.

Is this still something we want to do?

**Review by grahamalama — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (leplatrem) [reply README.rst:193]: I know that we don't want to do this anymore, but I felt obliged to reflect the current state of the repo.

Comment (grahamalama) [reply README.rst:193]: Makes sense. Can we create an issue to track the work for removing this step? Or do we already have one?

Comment (leplatrem) [reply README.rst:193]: I opened https://github.com/mozilla-services/kinto-dist/issues/1960 :ok_hand:

**Pull request merged**